### DISCUSSION TITLE
Show the table content and query when selecting the table

### DISCUSSION THREAD

**Discussion by leonardyhuang** [getredash/redash#6442] [Ideas]
Hi everyone,

I am not entirely sure if this is a duplicate request, I tried to search but I don't seem to find a similar one.

So basically, the idea is to show these 2 items when clicking the table name:
1.  Table content/records, with limit
2. The query, `select * from <table_name> limit X`

Thank you.

**Comment by guidopetri** [comment]
@leonardyhuang , thanks for the suggestion! Could you elaborate a bit more on why this would improve your workflow?

As it stands, it looks like we already get the table schema (columns + types) when we click on a table name on the left panel:
![image](https://github.com/getredash/redash/assets/18634426/2f47e748-a23f-49dc-bc79-0f1c00970132)

**Reply by leonardyhuang** [reply]
Hi @guidopetri, thanks for the prompt reply.

So, this feature actually is available in some DB editors and I thought it is cool to have on redash too. When you click/double-click on a table name, it will straightaway show the data (ie: 50 records by default) and also create the query `select * from <table_name>;` on the SQL editor. And, this helps our engineers to reduce the need to manually type the select query if they just want to quickly see some data samples.

**Reply by zachliu** [reply]
@guidopetri ~how come your redash is able to show the types? i'm playing with the `master` branch right now (using AWS Athena datasource) and got no types displaying on the UI~ depending on the `query_runner` your datasource is using, some of them support displaying data types some of them don't
